trestlebot/entrypoints/init.py

```diff
@@ -0,0 +1,207 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright (c) 2024 Red Hat, Inc.
+
+
+"""
+This module provides the `init` entrypoint for trestlebot.
+
+The init command is used for creating a new trestlebot workspace.  The command
+will do the following:
+    - create the .trestlebot directory for trestlebot related files
+    - create files for use with the specified CICD provider
+    - create the appropriate directories for the specified OSCAL model
+    - run the trestle init command to generate the needed compliance-trestle directories
+"""
+
+import argparse
+import importlib.resources
+import logging
+import pathlib
+import shutil
+import sys
+import traceback
+
+from trestle.common import file_utils
+from trestle.core.commands.init import InitCmd
+
+from trestlebot.const import (
+    ERROR_EXIT_CODE,
+    GITHUB,
+    GITHUB_WORKFLOWS_DIR,
+    GITLAB,
+    SUCCESS_EXIT_CODE,
+    TRESTLEBOT_CONFIG_DIR,
+    TRESTLEBOT_KEEP_FILE,
+)
+from trestlebot.entrypoints.entrypoint_base import EntrypointBase, handle_exception
+from trestlebot.entrypoints.log import set_log_level_from_args
+from trestlebot.tasks.authored import types as model_types
+
+
+logger = logging.getLogger(__name__)
+logging.getLogger("trestle.core.commands.init").setLevel("CRITICAL")
+
+OSCAL_MODEL_SSP = model_types.AuthoredType.SSP.value
+OSCAL_MODEL_COMPDEF = model_types.AuthoredType.COMPDEF.value
+
+
+class InitEntrypoint(EntrypointBase):
```

Comment:
Wondering if this should not inherit from EntrypointBase and if we should put this command in a different location given that it does not share the same core logic as the others. I think the only flags `init` would need to duplicate is `verbose` and `workspace`. The `dry-run` flag does not appear to be used.

Reply:
Yes, that sounds good to me.